### PR TITLE
docs/reference/glossary.rst: New terms, reduce complexity.

### PR DESCRIPTION
**Pull request by jimmo**

### CODE REVIEW

File: docs/reference/glossary.rst [hunk 1]
```diff
@@ -4,152 +4,190 @@ Glossary
...
-        a tuple is protected from this.)
+        This is a MicroPython-specific construct where, for efficiency
+        reasons, some built-in functions or methods may re-use the same
+        underlying tuple object to return data. This avoids having to allocate
+        a new tuple for every call, and reduces heap fragmentation. Programs
+        should not hold references to callee-owned tuples and instead only
+        extract data from them (or make a copy).
+
+    CircuitPython
+        An alternate version of MicroPython developed by `Adafruit Industries
```
Comment (stinos) [inline docs/reference/glossary.rst]: I think `alternative` might be more correct here?

File: docs/reference/glossary.rst [hunk 2]
```diff
@@ -4,152 +4,190 @@ Glossary
...
-        are intended to be installed individually - either using manual
-        copying or using :term:`upip`.
+        library comparable with :term:`CPython`'s. Instead, there is a related,
+        but separate project `micropython-lib
+        <https://github.com/micropython/micropython-lib>`_ which provides
+        implementations for many modules from CPython's standard library.
+        However, a large subset of these modules require POSIX-like
+        environment (Linux, FreeBSD, MacOS, etc.; Windows may be partially
+        supported), and thus would work or make sense only on the `MicroPython
+        Unix port`. Some subset of modules are usable for the
```
Comment (stinos) [inline docs/reference/glossary.rst]: Maybe add a short explanation that the main reason for this (i.e. most of it only being usable on POSIX-like) is that's it is implemented by ffi?

File: docs/reference/glossary.rst [hunk 3]
```diff
@@ -4,152 +4,190 @@ Glossary
...
+        The unix port is one of the major :term:`MicroPython ports
+        <MicroPython port>`. It is intended to run on POSIX-compatible
+        operating systems, like Linux, MacOS, FreeBSD, Solaris, etc. It also
+        serves as the basis of Windows port. The Unix port is very useful for
+        quick development and testing of the MicroPython language and
+        machine-independent features. It can also function in a similar way to
+        :term:`CPython`'s ``python`` executable.
+
+    .mpy file
+        The output of the cross compiler. A compiled form of a :term:`.py` file
```
Comment (stinos) [inline docs/reference/glossary.rst]: Also use `:term:` for 'cross compiler'?

**Comment by jimmo** [comment]
> I think alternative might be more correct here?

You're absolutely right. However, I changed it to "A variant of MicroPython" which I think reads better.

Fixed the other comments too. Thanks!

**Comment by mcauser** [comment]
Looks good. Found `acryonym` misspelled twice further down.
"acryonym" -> "acronym"

**Comment by mcauser** [comment]
In upip: "A package manage" -> "A package manager"

**Comment by jimmo** [comment]
> "acryonym" -> "acronym"
> In upip: "A package manage" -> "A package manager"

Fixed. Thanks!

**Comment by mattytrentini** [comment]
These descriptions are clearer to me @jimmo, thanks!

**Comment by dpgeorge** [comment]
Thanks a lot!  Merged in a93495b66d1e1101fffa3c8f2811d8750b5601f9 with some minor fixes to the cross references within the glossary itself.

**Pull request closed without merging**